File: Renderer/src/types/instructions.ts
enum InstructionType {
    String,
    Rectangle
}

interface Instruction {
    type: InstructionType;
    data: any;
}

interface StringInstructionData {
    content: string;
    font: string;
    color: number;
    x: number;
    y: number;
    size: number;
}

interface RectangleInstructionData {
    x: number;
    y: number;
    width: number;
    height: number;
    color: number;
    fill: boolean;
    thickness: number;
    radius: number;
    alpha: number;
}

export {
    InstructionType,
    Instruction,
    StringInstructionData,
    RectangleInstructionData
}

File: Discover/Types/Instructions.cs
namespace Discover.Types
{
    public class Instructions
    {
        public enum InstructionType
        {
            String,
            Rectangle
        }

        public struct Instruction
        {
            public InstructionType Type { get; set; }
            public object Data { get; set; }
        }

        public struct StringInstructionData
        {
            public string Content { get; set; }
            public string Font { get; set; }
            public int Color { get; set; }
            public int X { get; set; }
            public int Y { get; set; }
            public int Size { get; set; }
        }

        public struct RectangleInstructionData
        {
            public int X { get; set; }
            public int Y { get; set; }
            public int Width { get; set; }
            public int Height { get; set; }
            public int Color { get; set; }
            public bool Fill { get; set; }
            public int Thickness { get; set; }
            public int Radius { get; set; }
        }
    }
}
